watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -330,14 +362,41 @@ def build(self):
             units=pyunits.m,
             doc="Stokes radius of solute",
         )
-        self.diffus_phase_comp = Param(
-            self.phase_list,
-            self.ion_set | self.solute_set,
+        if self.config.diffus_calculation == DiffusivityCalculation.none:
+            self.diffus_phase_comp = Param(
+                self.phase_list,
+                self.ion_set | self.solute_set,
+                mutable=True,
+                default=1e-9,
+                initialize=self.config.diffusivity_data,
+                units=pyunits.m**2 * pyunits.s**-1,
+                doc="Bulk diffusivity of solute components",
+            )
+        elif self.config.diffus_calculation == DiffusivityCalculation.HaydukLaudie:
+            hl_ind = [i[1] for i in self.config.molar_volume_data.keys()]
+            nonhl_ind = [i[1] for i in self.config.diffusivity_data.keys()]
+            if not (hl_ind + nonhl_ind) == self.ion_set | self.solute_set:
+                raise ConfigurationError(
+                    "Insufficient data is provided to determine diffusivity of all solute components. "
+                    "Check the molar_volume_data and diffusivity_data configuration."
+                )
+            self.diffus_phase_comp_nonhl = Param(
```

Comment:
Why give this a different name?

Reply:
Nevermind- I see you later specify the variable in the on-demand method.

Reply:
A warning for this however - IDAES v2.0 will be a lot more strict about objecting to non-standard naming of properties. You will have to explicitly declare non-standard properties as custom properties once you update to v2.0.

See line 2175 for an example: https://github.com/IDAES/idaes-pse/blob/main/idaes/models/properties/general_helmholtz/helmholtz_functions.py